File: solutions/java/src/cricinfo/Over.java
package cricinfo;

import java.util.ArrayList;
import java.util.List;

public class Over {
    private final int overNumber;
    private final List<Ball> balls;

    public Over(int overNumber) {
        this.overNumber = overNumber;
        this.balls = new ArrayList<>();
    }

    public void addBall(Ball ball) {
        this.balls.add(ball);
    }

    public int getOverNumber() {
        return overNumber;
    }

    public List<Ball> getBalls() {
        return balls;
    }
}

File: solutions/python/cricinfo/over.py
class Over:
    def __init__(self, over_number):
        self.over_number = over_number
        self.balls = []

    def add_ball(self, ball):
        self.balls.append(ball)

    def get_over_number(self):
        return self.over_number

    def get_balls(self):
        return self.balls
